### PR TITLE
Feature/105 full text tab search

### LINKED ISSUE #105: Feature Request: Make full page text searchable
I'd love to be able to search against the full html text in an open tab. Is there any way to do this in the product today? Do you have thoughts to make this feature available in the future?

Thanks, 

Niz

### PR DESCRIPTION
**Pull request by crossan007**
closes #105 

Adds full-text-search for all open tabs.

image

### CODE REVIEW

**Comment by crossan007** [comment]
@babyman Any thoughts on this?   I've been using it daily now, and performance seems to be good with even up to 70 open tabs.

I do still want to refine the search so it only displays ___rendered___ text, as opposed to searching the full source.

File: quick-tabs/popup.js [hunk 1]
```diff
@@ -453,9 +467,12 @@ function renderTabs(params, delay, currentTab) {
     'noResults': allTabs.length === 0 && closedTabs.length === 0 && bookmarks.length === 0 && history.length === 0,
     'hasClosedTabs': closedTabs.length > 0,
     'hasBookmarks': bookmarks.length > 0,
+    'hasFullTextTabs': fullTextTabs.length > 0,
     'hasHistory': history.length > 0
   };
 
+  console.log(context);
```
Comment (Rohaq) [inline quick-tabs/popup.js]: Remove unneeded console.log line.

File: quick-tabs/popup.js [hunk 2]
```diff
@@ -629,15 +646,56 @@ AbstractSearch.prototype.executeSearch = function(query) {
...
+                {
+                  tab.snippet=result[0].snippet;
+                  window.RenderParams.fullTextTabs.push(tab);
+                }
+                
+                
+              });
+            }
+            window.tabsDone ++;
+            console.log("Tabs Done: " +window.tabsDone + "/"+window.tabsLeft+" for query: "+window.query);
```
Comment (Rohaq) [inline quick-tabs/popup.js]: Remove unneeded console.log line.

**Comment by crossan007** [comment]
@buckleyGI I'm not sure the answer to that ;-)

Last I committed to this, it was working; however it was never approved / merged by the project owner.

**Comment by JoelAzaria** [comment]
Please ensure that if this is included it's optional.  I really don't think I'd need this on at all times (actually I would probably be bothered by it being on constantly)

Not to be a wet blanket - Just want to make sure I can have it off.  
If there was an on/off toggle right by the search, maybe I'd use it on some limited occasions.

**Comment by JayGoldberg** [comment]
I know that you worked hard on this, but the necessity to change permissions to "Read and change all your data on the websites you visit" is a dealbreaker for me.

I believe corporate Chrome policy allows blocking of extensions based on this permission. For sure it allows shadow-blocking of access to the DOM for certain web properties by URL pattern match, which at best would disable the functionality implemented here (it just wouldn't work) or at worst, break the extension entirely if it doesn't expect to be blocked access to the DOM.

Many users are just skeptical of any extension that requests "Read and change all your data on the websites you visit", it's a very generous and dangerous permission.

**Pull request closed without merging**